validator/src/main/java/com/google/daq/mqtt/util/ConfigManager.java

```diff
@@ -275,4 +279,36 @@ public String getUpdatedTimestamp() {
     return isoConvert(metadata.timestamp);
   }
 
+  private DiscoveryConfig getDiscoveryConfig() {
+    if (metadata.discovery == null) {
+      return null;
+    }
+
+    DiscoveryConfig discoveryConfig = new DiscoveryConfig();
+    discoveryConfig.families = ifNotNullGet(metadata.discovery.families,
+        families -> new HashMap<>());
+
+    if (discoveryConfig.families != null) {
+      metadata.discovery.families.keySet().forEach(family -> {
+        FamilyDiscoveryModel familyDiscoveryModel = metadata.discovery.families.get(family);
+        if (familyDiscoveryModel != null) {
+          // for a sporadic scan, supplying generation is invalid
+          if (familyDiscoveryModel.scan_interval_sec == null) {
```

Comment:
This can be simplified into one logical expression (rather than the if- and checkState) -- checkState(scan_interval_sec != null || generation == null, "xxxx")